### PR TITLE
Destination S3: Add virtual-hosted-style option

### LINKED ISSUE #19426: Destination S3: add virtual-hosted-style option
## Tell us about the problem you're trying to solve
I need to add Aliyun OSS(S3 compatible) as my destination, which only support virtual hosted style endpoint.

## Describe the solution you’d like
Provider an option for S3 destination to enable/disable virtual hosted style.

## Are you willing to submit a PR?
Yes, I will submit a PR.

### PR DESCRIPTION
**Pull request by Xingyuan-Chen**
*I add a virtual-hosted-style option for S3 destination
*It helps to make S3 destination support Non-Amazon S3 endpoint who only support virtual hosted style like Aliyun OSS.
This PR closes https://github.com/airbytehq/airbyte/issues/19426

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java [hunk 1]
```diff
@@ -380,9 +399,12 @@ public S3DestinationConfig get() {
...
-          uploadThreadsCount);
+          uploadThreadsCount,
+          pathStyleAccessEnabled);
     }
 
   }
 
 }
+
+389
```
Comment (marcosmarxm) [inline airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java:410]: Please update this

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java (hunk 1 above)
Comment (Xingyuan-Chen) [reply airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java]: updated

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by dtonda** [comment]
Note that this may also solve issue #20860 as it would provide an alternative way to configure the S3 connection for non-AWS S3 buckets.

**Comment by ZengineChris** [comment]
Any updates on this?

**Review by grishick — CHANGES_REQUESTED** [review]
Several comments on this PR:
1. This PR does not compile, because `PATH_STYLE_ACCESS_ENABLED` is not defined.
2. `isPathStyleAccessEnabled` getter method is not used anywhere
3. `S_3_PATH_STYLE_ACCESS_ENABLED` is defined but not used anywhere
4. In order for this feature to be usable, this PR should add a new field to `spec.json `to `destination-s3` connector

**Comment by marcosmarxm** [comment]
Closed due inactivity @Xingyuan-Chen let us know if you want to return to this PR.

**Pull request closed without merging**